main.js

```diff
@@ -796,10 +796,20 @@ addStealExtension(function applyTraceExtension(loader) {
 // Steal JSON Format
 // Provides the JSON module format definition.
 addStealExtension(function (loader) {
-  var jsonTest = /^[\s\n\r]*[\{\[]/;
   var jsonExt = /\.json$/i;
   var jsExt = /\.js$/i;
 
+  // taken from prototypejs
+  // https://github.com/sstephenson/prototype/blob/master/src/prototype/lang/string.js#L682-L706
+  function isJSON(str) {
+    if (!str) return false;
+
+    str = str.replace(/\\(?:["\\\/bfnrt]|u[0-9a-fA-F]{4})/g, '@');
```

Comment:
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp)

```
The literal notation provides compilation of the regular expression when the expression is evaluated. Use literal notation when the regular expression will remain constant. For example, if you use literal notation to construct a regular expression used in a loop, the regular expression won't be recompiled on each iteration.
```

also http://stackoverflow.com/questions/8814009/how-often-does-javascript-recompile-regex-literals-in-functions

I think we are good there, want me to move them outside anyway?